### PR TITLE
FIX #208, getFacetValues with no values for the facet should not throw.

### LINKED ISSUE #208: getFacetValues throws when no hits returned
When a search returns no hits, calling `getFacetValues` for an existing facet will throw in the `extractNormalizedFacetValues ` function.

At [this line](https://github.com/algolia/algoliasearch-helper-js/blob/master/src/SearchResults/index.js#L434), `facet` is undefined because the `results.facets` array is empty, so trying to access `data` on `facet` raises an error:

```javascript
    if (results._state.isConjunctiveFacet(attribute)) {
        var facet = find(results.facets, predicate);
        var facetValues = map(facet.data, function(v, k) { // Uncaught TypeError: Cannot read property 'data' of undefined
```

I guess the problem would arise on lines [445](https://github.com/algolia/algoliasearch-helper-js/blob/master/src/SearchResults/index.js#L445) and [456](https://github.com/algolia/algoliasearch-helper-js/blob/master/src/SearchResults/index.js#L456) as well!

While it's easy enough to check for `nbHits` before calling `getFacetValues`, I guess it'd be better to safeguard the function internally :)

### PR DESCRIPTION
**Pull request by bobylito**
Instead it should gracefully return an empty array (the behavior was
correctly defined for the hierarchical facets)

### CODE REVIEW

**Comment by bobylito** [comment]
@redox

**Comment by redox** [comment]
LGTM :+1:

File: src/SearchResults/index.js [hunk 1]
```diff
@@ -420,6 +420,8 @@ SearchResults.prototype.getFacetByName = function(name) {
     find(this.hierarchicalFacets, predicate);
 };
 
+var EMPTY_NORMALIZED_FACET_VALUES = (function() { var a = []; a.facetType = 'conjunctive'; return a; })();
```
Comment (vvo) [inline src/SearchResults/index.js]: Why do we need to attach a property containing `facetType` to the array here? why not just return an empty array?

Comment (vvo) [inline src/SearchResults/index.js]: Also what do you think of:

```js
var EMPTY_NORMALIZED_FACET_VALUES = [];
EMPTY_NORMALIZED_FACET_VALUES.facetType = 'conjunctive'
``

"Ugly" But easy to read.

Comment (vvo) [inline src/SearchResults/index.js]: > Why do we need to attach a property containing facetType to the array here? why not just return an empty array?

Ok I saw the other lines in the helper, I did not know we had this in the helper. Is it something we use in some implementations? (facetType)

Comment (bobylito) [inline src/SearchResults/index.js]: Actually, after reading the code, this property is only necessary for case it's a hierarchical facet. We may remove it :)

Comment (vvo) [inline src/SearchResults/index.js]: Damn hierarchicalFacets! Who did this!!? :dancer:

**Comment by vvo** [comment]
Lint did not pass

**Comment by bobylito** [comment]
Fixing :)

File: src/SearchResults/index.js (hunk 1 above)
Comment (bobylito) [inline src/SearchResults/index.js]: But the documentation states that this values is attached when returned from extractNormalizedFacetValues, but then we should probably remove it from the returned array.

Comment (vvo) [inline src/SearchResults/index.js]: Did not even know about extractNormalizedFacetValues O_O

**Comment by vvo** [comment]
FLGTM

**Comment by bobylito** [comment]
Everything is alright then, merging!

**Pull request merged**